packages/lsp/src/tokens.ts

```diff
@@ -851,6 +856,20 @@ export interface ILSPConnection extends ILspConnection, IObservableDisposable {
    */
   errorSignal: ISignal<ILSPConnection, any>;
 
+  /**
+   * @alpha
+   *
+   * Check if a capability is available in the server capabilities.
+   */
+  provides(capability: keyof lsp.ServerCapabilities): boolean;
+
+  /**
+   * @alpha
+   *
+   * Lists server capabilities.
+   */
+  serverCapabilities: lsp.ServerCapabilities;
```

Comment:
Should it be `readonly`?

Reply:
I asked myself the same question. Possibly yes. But none of the APIs which are read-only in implementation are marked as such in the interface. I am not sure why.